src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -110,6 +110,87 @@ public int Count
             }
         }
 
+        public int Length => Count;
+
+        public T this[int index]
+        {
+            get
+            {
+                if (_array == null)
+                    ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_NullArray);
+                if (index < 0 || index >= _count)
+                    ThrowHelper.ThrowArgumentOutOfRange_IndexException();
+                Contract.EndContractBlock();
+
+                return _array[_offset + index];
+            }
+
+            set
+            {
+                if (_array == null)
+                    ThrowHelper.ThrowInvalidOperationException(ExceptionResource.InvalidOperation_NullArray);
+                if (index < 0 || index >= _count)
+                    ThrowHelper.ThrowArgumentOutOfRange_IndexException();
+                Contract.EndContractBlock();
+
+                _array[_offset + index] = value;
+            }
+        }
+
+        /// <summary>
+        /// Returns an empty <see cref="ArraySegment{T}"/>
+        /// </summary>
+        public static ArraySegment<T> Empty
+        {
+            get { return default(ArraySegment<T>); }
+        }
+
+        public ArraySegment<T> Slice(int start) => Slice(start, _count - start);
+
+        public ArraySegment<T> Slice(int start, int length)
+        {
+            if (_array == null)
```

Comment:
I can describe it like next
1. `(uint)start > (uint)_count` equal to `start < 0 or start > _count`
2. `(uint)length > (uint)(_count - start))` equal to `length < 0 or length > _count - start` where `_count - start >= 0` is checked at step 1.